File: CS/SilverlightApplication1.Web/SchedulingDomainService.metadata.cs

namespace SilverlightApplication1.Web {
    using System;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.ComponentModel.DataAnnotations;
    using System.Linq;
    using System.ServiceModel.DomainServices.Hosting;
    using System.ServiceModel.DomainServices.Server;


    // The MetadataTypeAttribute identifies CarSchedulingMetadata as the class
    // that carries additional metadata for the CarScheduling class.
    [MetadataTypeAttribute(typeof(CarScheduling.CarSchedulingMetadata))]
    public partial class CarScheduling {

        // This class allows you to attach custom attributes to properties
        // of the CarScheduling class.
        //
        // For example, the following marks the Xyz property as a
        // required property and specifies the format for valid values:
        //    [Required]
        //    [RegularExpression("[A-Z][A-Za-z0-9]*")]
        //    [StringLength(32)]
        //    public string Xyz { get; set; }
        internal sealed class CarSchedulingMetadata {

            // Metadata classes are not meant to be instantiated.
            private CarSchedulingMetadata() {
            }

            public bool AllDay { get; set; }

            public Nullable<int> CarId { get; set; }

            public string ContactInfo { get; set; }

            public string Description { get; set; }

            public Nullable<DateTime> EndTime { get; set; }

            public Nullable<int> EventType { get; set; }

            public int ID { get; set; }

            public Nullable<int> Label { get; set; }

            public string Location { get; set; }

            public Nullable<decimal> Price { get; set; }

            public string RecurrenceInfo { get; set; }

            public string ReminderInfo { get; set; }

            public Nullable<DateTime> StartTime { get; set; }

            public Nullable<int> Status { get; set; }

            public string Subject { get; set; }

            public Nullable<int> UserId { get; set; }
        }
    }
}

File: VB/SilverlightApplication1.Web/SchedulingDomainService.metadata.vb

Option Compare Binary
Option Infer On
Option Strict On
Option Explicit On

Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.ComponentModel.DataAnnotations
Imports System.Linq
Imports System.ServiceModel.DomainServices.Hosting
Imports System.ServiceModel.DomainServices.Server


'The MetadataTypeAttribute identifies CarSchedulingMetadata as the class
' that carries additional metadata for the CarScheduling class.
<MetadataTypeAttribute(GetType(CarScheduling.CarSchedulingMetadata))>  _
Partial Public Class CarScheduling
    
    'This class allows you to attach custom attributes to properties
    ' of the CarScheduling class.
    '
    'For example, the following marks the Xyz property as a
    ' required property and specifies the format for valid values:
    '    <Required()>
    '    <RegularExpression("[A-Z][A-Za-z0-9]*")>
    '    <StringLength(32)>
    '    Public Property Xyz As String
    Friend NotInheritable Class CarSchedulingMetadata
        
        'Metadata classes are not meant to be instantiated.
        Private Sub New()
            MyBase.New
        End Sub
        
        Public Property AllDay As Boolean
        
        Public Property CarId As Nullable(Of Integer)
        
        Public Property ContactInfo As String
        
        Public Property Description As String
        
        Public Property EndTime As Nullable(Of DateTime)
        
        Public Property EventType As Nullable(Of Integer)
        
        Public Property ID As Integer
        
        Public Property Label As Nullable(Of Integer)
        
        Public Property Location As String
        
        Public Property Price As Nullable(Of Decimal)
        
        Public Property RecurrenceInfo As String
        
        Public Property ReminderInfo As String
        
        Public Property StartTime As Nullable(Of DateTime)
        
        Public Property Status As Nullable(Of Integer)
        
        Public Property Subject As String
        
        Public Property UserId As Nullable(Of Integer)
    End Class
End Class

